R/utils.R

```diff
@@ -228,22 +217,27 @@ get_data_type_to_data_model_mapping <- function() {
 #' @keywords utils
 #' @export
 data_model.character <- function(x) {
-  # TODO: switch to gDRutils::get_experiment_groups()
-  # once we clean-up single-agent/combination assignemnts
-  exp_v <- get_data_type_to_data_model_mapping()
-  
-  checkmate::assert_subset(x, names(exp_v))
-  
-  exp_v[[x]]
+  checkmate::assert_subset(x, gDRutils::get_supported_experiments())
   
+  exp_v <- gDRutils::get_experiment_groups()
+  names(exp_v[grep(x, exp_v)])
 }
 
+
+#' Validate availability of data models
+#' 
+#' @param d_types character vector with experiment names in \code{MultiAssayExperiment} object
+#' @param s_d_models character vector with names of supported experiment
+#'
+#' @keywords internal
 validate_data_models_availability <- function(d_types, s_d_models) {
+  checkmate::assert_character(d_types)
+  checkmate::assert_character(s_d_models, null.ok = TRUE)
   
-  dm_v <- get_data_type_to_data_model_mapping()
-  
+  dm_v <- gDRutils::get_experiment_groups()
+
   req_d_models <-
-    unique(as.character(dm_v[names(dm_v) %in% d_types]))
+    unique(names(dm_v)[vapply(d_types, function(i) grep(i, dm_v), integer(1))])
```

Comment:
Is the `grep` needed here?

Reply:
@gladkia  Could you proposed other solution?

Reply:
```
exp <- gDRutils::get_supported_experiments()
exp
dm_v <- gDRutils::get_experiment_groups()
dm_v

d_types <- exp[1]
d_types
req_d_models <-
  unique(names(dm_v)[vapply(d_types, function(i) grep(i, dm_v), integer(1))])
req_d_models

d_types <- c(exp[2], exp[3])
req_d_models <-
  unique(names(dm_v)[vapply(d_types, function(i) grep(i, dm_v), integer(1))])
req_d_models

d_types <- c(exp[1], exp[3])
req_d_models <-
  unique(names(dm_v)[vapply(d_types, function(i) grep(i, dm_v), integer(1))])
req_d_models
```

Reply:
@j-smola you can get rid of grep by using e.g. `unique(names(dm_v)[vapply(dm_v, function(x) any(d_types %in% x), logical(1))])`

Reply:
or shorter without `vapply` `unique(names(Filter(function(x) any(d_types %in% x), dm_v)))`

Reply:
Changed into `any` :)